packages/ListBox/src/hooks/hooks.js

```diff
@@ -149,17 +160,6 @@ export function useOnScrolled() {
   }, [state.activeOption, state.options, state.refListBox]);
 }
 
-export function useIsDisabled(isDisabled) {
-  const [, dispatch] = useListBox();
-
-  React.useEffect(() => {
-    dispatch({
-      type: useListBox.types.setIsDisabled,
-      payload: isDisabled,
-    });
-  }, [dispatch, isDisabled]);
```

Comment:
This seems like a complex way to keep a consumer-provided prop in sync with the component's internal state and what motivated me to find a new pattern for the new `isReadOnly` and `hasError` props.